client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -304,6 +334,29 @@ private void write01(ReaderGroupConfig object, RevisionDataOutput revisionDataOu
             revisionDataOutput.writeMap(object.endingStreamCuts, keySerializer, valueSerializer);
             revisionDataOutput.writeInt(object.getMaxOutstandingCheckpointRequest());
         }
+
+        private void read02(RevisionDataInput revisionDataInput, ReaderGroupConfigBuilder builder) throws IOException {
```

Comment:
This logic is incorrect. You don't need to re-serialize all the previous fields.

Reply:
Not setting those values causes this test case to fail:
```
io.pravega.client.stream.impl.SerializationTest > testReaderGroupInit FAILED
 java.lang.AssertionError: expected:<ReaderGroupState.CompactReaderGroupState(config=ReaderGroupConfig(groupRefreshTimeMillis=261, automaticCheckpointIntervalMillis=-1, startingStreamCuts={StreamImpl(scope=MnPkB, streamName=Klytg)=UNBOUNDED}, endingStreamCuts={StreamImpl(scope=MnPkB, streamName=Klytg)=UNBOUNDED}, maxOutstandingCheckpointRequest=3, retentionConfig=NONE), checkpointState=CheckpointState { ongoingCheckpoints: [UqObC, dnUVN],  readersBlockingEachCheckpoint: {jEdbi=[]} }, distanceToTail={crzYB=-2509738709635323033}, futureSegments={}, assignedSegments={uiHLL={}}, unassignedSegments={SegmentWithRange(segment=pUzSq/jfyqa/65.#epoch.0, range=SegmentWithRange.Range(low=0.06767117977142334, high=0.7583411931991577))=-8086877361167159070, SegmentWithRange(segment=xGabw/OmZmb/21.#epoch.0, range=SegmentWithRange.Range(low=0.5213870406150818, high=0.6767080426216125))=1153959188627795253}, lastReadPosition={SegmentWithRange(segment=vknHM/DqlWO/28.#epoch.0, range=SegmentWithRange.Range(low=0.16790086030960083, high=0.20448356866836548))=9050953671715088000}, endSegments={uSSnW/OGpLi/75.#epoch.0=3759285871800690071, MYTxV/xCRdJ/18.#epoch.0=-5404048114723376066})> but was:<ReaderGroupState.CompactReaderGroupState(config=ReaderGroupConfig(groupRefreshTimeMillis=261, automaticCheckpointIntervalMillis=-1, startingStreamCuts={StreamImpl(scope=MnPkB, streamName=Klytg)=UNBOUNDED}, endingStreamCuts={StreamImpl(scope=MnPkB, streamName=Klytg)=UNBOUNDED}, maxOutstandingCheckpointRequest=3, retentionConfig=NONE), checkpointState=CheckpointState { ongoingCheckpoints: [UqObC, dnUVN],  readersBlockingEachCheckpoint: {jEdbi=[]} }, distanceToTail={crzYB=-2509738709635323033}, futureSegments={}, assignedSegments={uiHLL={}}, unassignedSegments={SegmentWithRange(segment=xGabw/OmZmb/21.#epoch.0, range=SegmentWithRange.Range(low=0.5213870406150818, high=0.6767080426216125))=1153959188627795253, SegmentWithRange(segment=pUzSq/jfyqa/65.#epoch.0, range=SegmentWithRange.Range(low=0.06767117977142334, high=0.7583411931991577))=-8086877361167159070}, lastReadPosition={SegmentWithRange(segment=vknHM/DqlWO/28.#epoch.0, range=SegmentWithRange.Range(low=0.16790086030960083, high=0.20448356866836548))=9050953671715088000}, endSegments={MYTxV/xCRdJ/18.#epoch.0=-5404048114723376066, uSSnW/OGpLi/75.#epoch.0=3759285871800690071})>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at io.pravega.client.stream.impl.SerializationTest.verify(SerializationTest.java:330)
        at io.pravega.client.stream.impl.SerializationTest.testReaderGroupInit(SerializationTest.java:266)
```

Reply:
fixed.